From 245eb3018094f4c1f645283899d80c2b00e2c2d6 Mon Sep 17 00:00:00 2001
From: Diana Picus <diana.picus@linaro.org>
Date: Wed, 7 Sep 2016 08:37:15 +0000
Subject: [PATCH 4/8] [CMake] Use CMake's default RPATH for the unit tests

In the top-level CMakeLists.txt, we set CMAKE_BUILD_WITH_INSTALL_RPATH to ON,
and then for the unit tests we set it to <test>/../../lib. This works for tests
that live in unittest/<whatever>, but not for those that live in subdirectories
e.g. unittest/Transforms/IPO or unittest/ExecutionEngine/Orc. When building
with BUILD_SHARED_LIBRARIES, such tests don't manage to find their libraries.

Since the tests are run from the build directory, it makes sense to set their
RPATH for the build tree, rather than the install tree. This is the default in
CMake since 2.6, so all we have to do is set CMAKE_BUILD_WITH_INSTALL_RPATH to
OFF for the unit tests.

git-svn-id: https://llvm.org/svn/llvm-project/llvm/trunk@280791 91177308-0d34-0410-b5e6-96231b3b80d8
(cherry picked from commit 1317eef21cdf7989d233a6f08322283a217eeb20)
---
 unittests/CMakeLists.txt | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git llvm_release_39/unittests/CMakeLists.txt macports_release_39/unittests/CMakeLists.txt
index 49a9b31b60d..1758d2b2704 100644
--- llvm_release_39/unittests/CMakeLists.txt
+++ macports_release_39/unittests/CMakeLists.txt
@@ -1,11 +1,9 @@
 add_custom_target(UnitTests)
 set_target_properties(UnitTests PROPERTIES FOLDER "Tests")
 
-if (APPLE)
-  set(CMAKE_INSTALL_RPATH "@executable_path/../../lib")
-else(UNIX)
-  set(CMAKE_INSTALL_RPATH "\$ORIGIN/../../lib${LLVM_LIBDIR_SUFFIX}")
-endif()
+# People tend to run the tests _before_ installing, so we don't want the install
+# rpath here.
+set(CMAKE_BUILD_WITH_INSTALL_RPATH OFF)
 
 function(add_llvm_unittest test_dirname)
   add_unittest(UnitTests ${test_dirname} ${ARGN})
-- 
2.11.0

